system_tests/pubsub.py

```diff
@@ -169,15 +170,24 @@ def test_message_pull_mode_e2e(self):
         topic.publish(MESSAGE_1, extra=EXTRA_1)
         topic.publish(MESSAGE_2, extra=EXTRA_2)
 
-        received = subscription.pull(max_messages=2)
-        ack_ids = [recv[0] for recv in received]
-        subscription.acknowledge(ack_ids)
-        messages = [recv[1] for recv in received]
+        class Hoover(object):
 
-        def _by_timestamp(message):
-            return message.timestamp
+            def __init__(self):
+                self.received = []
 
-        message1, message2 = sorted(messages, key=_by_timestamp)
+            def done(self, *dummy):
+                return len(self.received) == 2
+
+            def suction(self):
+                with subscription.auto_ack(max_messages=2) as ack:
+                    self.received.extend(ack.values())
+
+        hoover = Hoover()
+        retry = RetryInstanceState(hoover.done)
+        retry(hoover.suction)()
```

Comment:
Because we auto-ack the responses, a retry will not fetch an messages which had already been pulled, so we are purely accumulating them.